include/reactor-uc/network_channel.h

```diff
@@ -31,14 +31,27 @@ typedef enum {
 typedef enum {
   NETWORK_CHANNEL_TYPE_TCP_IP,
   NETWORK_CHANNEL_TYPE_COAP_UDP_IP,
+  NETWORK_CHANNEL_TYPE_I2C,
+  NETWORK_CHANNEL_TYPE_UART
 } NetworkChannelType;
 
+typedef enum {
+  NETWORK_CHANNEL_CATEGORY_ASYNC,
+  NETWORK_CHANNEL_CATEGORY_SYNC,
+} NetworkChannelCategory;
```

Comment:
I think `Category` is a little too vague. What about `Mode`, e.g. `NetworkChannelMode`. But since we only have two modes, we could also just use a boolean flag, like: `boolean polled`

Reply:
done